code/ATMOSPHERICS/atmospherics.dm

```diff
@@ -348,14 +348,17 @@ Pipelines + Other Objects -> Pipe network
 
 	var/obj/machinery/atmospherics/target_move = findConnecting(direction, user.ventcrawl_layer)
 	if(target_move)
-		if(is_type_in_list(target_move, ventcrawl_machinery) && target_move.can_crawl_through())
+		if(is_type_in_list(target_move, ventcrawl_machinery) && target_move.can_crawl_through() && !user.ventcrawl_exit_target)
 			user.visible_message("Something is squeezing through the ducts...", "You start crawling out the ventilation system.")
 			target_move.shake(2, 3)
-			if(do_after(user, target_move, 10))
-				user.remove_ventcrawl()
-				user.forceMove(target_move.loc) //handles entering and so on
-				user.visible_message("You hear something squeeze through the ducts.", "You climb out the ventilation system.")
-		else if(target_move.can_crawl_through())
+			user.ventcrawl_exit_target = target_move
+			spawn
```

Comment:
DM is stupid

Can you change it to spawn() so nobody else has a heart attack please